manager/controllers/app/m4dapplication_controller.go

```diff
@@ -381,3 +390,30 @@ func (r *M4DApplicationReconciler) GetAllModules() (map[string]*app.M4DModule, e
 	}
 	return moduleMap, nil
 }
+
+// GetProcessingGeography determines the geography of the workload cluster.
+// If no workload has been specified, a local cluster is assumed.
+func (r *M4DApplicationReconciler) GetProcessingGeography(applicationContext *app.M4DApplication) (string, error) {
+	clusterName := applicationContext.Spec.Selector.ClusterName
+	if clusterName == "" {
```

Comment:
Related to previous comment ... Why check for an empty string instead of nil?

Reply:
It is not nil. It will be an empty structure with all strings in it equal to an empty string.